### ISSUE TITLE
Save the model before over-fitting

### ISSUE THREAD

**Issue by dimstudio** [dimstudio/SharpFlow#11]
Right now the saved model is at 120th epoch.

**Comment by HansBambel** [comment]
The fit function takes a `save_every` argument that saves the model every e.g. `5` epochs. 

We could also implement (or use) an early stopping criterion.

**Comment by dimstudio** [comment]
`save_every` doesn't sound a solution to me. 

Digging in some old code I found this approach. What do you think?

```
a = model_history.history['loss']
b = model_history.history['val_loss']
d = [a_i - b_i for a_i, b_i in zip(a, b)]
linex = next(d.index(i) for i in reversed(d) if i >= 0) #overfitting epoch
```

**Comment by HansBambel** [comment]
But that would be after you have done training the model, right? So you still train the whole model and you haven't saved it until the end, right? Usually you would train until the validation loss does not improve for a certain amount of epochs and then stop.

Btw: You can read these values out with the tensorboard as well.

Another more pythonic way for the code would be (the zip is quite expensive if I recall correctly):
`d = model_history.history['loss'] - model_history.history['val_loss']`
`index = np.arange(len(d))[d >= 0.0][-1]`

**Comment by dimstudio** [comment]
> But that would be after you have done training the model, right? So you still train the whole model and you haven't saved it until the end, right? 

Yes I think that's an idea, we only do training once 

> Usually you would train until the validation loss does not improve for a certain amount of epochs and then stop.

OK that makes more sense, perhaps we don't even need to look in `training_loss`

> Another more pythonic way for the code would be (the zip is quite expensive if I recall correctly):

Ok thanks, sometimes my code doesn't shine :-)
